Update theme tier badges in Theme Details and Theme Showcase (LiTS & LoTS)

abotic opened the pull request:
Related to:
- https://github.com/Automattic/wp-calypso/issues/96162

## Proposed Changes
- Improved badge visibility in theme details page to match theme grid design
- Added clear hierarchy for badge display:
  - Shows `Free` badge for free themes when user is not logged in
  - Shows `Included with plan` badge for themes available in user's current plan
  - Shows `Available on {plan name}` for themes requiring higher tier plans
  - Shows `Available on Business` for WooCommerce themes
  - Shows `Available on Business (and price €X.XX/month where applicable)` for partner themes
  - Shows correct plan upgrades for `Bundled` and `Community` themes
- Updated badge positioning to appear above theme title
- Removed style variations from theme grid view
- Unified badge styling across theme showcase and details pages
- Refactor components with memoization and selector optimizations 

## Why are these changes being made?
Currently, the theme details page doesn't clearly indicate which plan is required for themes, leading to user confusion. This update improves clarity by:
1. Making badge visibility consistent between grid and details views
2. Clearly indicating when a theme is included in the user's current plan
3. Specifically showing which plan tier is needed (Personal, Premium, Business, etc.)
4. Providing clear pricing information for partner themes

## Testing Instructions
1. Test each scenario in both Theme Showcase grid and Theme Details page
2. Test with both logged-in and logged-out states where applicable

| Theme Type | Login State | Location | Expected Behavior |
|------------|------------|----------|-------------------|
| Active theme | Logged in | Grid | Shows "Active" badge |
| Free theme | Logged out | Grid & Details | Shows "Free" badge above title |
| Personal plan theme | Logged in (no plan) | Grid & Details | Shows "Available on Personal" |
| Premium plan theme | Logged in (no/lower plan) | Grid & Details | Shows "Available on Premium" |
| WooCommerce plan theme | Logged in (no/lower plan) | Grid & Details | Shows "Available on Business" |
| Partner theme | Logged in (no Business plan) | Grid & Details |  Shows "Available on Business" (and price €X.XX/month where applicable) |
| Partner theme | Logged in (Business plan) | Grid & Details | Shows "€X.XX/month" |
| Bundled theme | Logged in (no/lower plan) | Grid & Details | Shows "Available on Business" |
| Bundled theme | Logged in (Business plan) | Grid & Details | Shows "Included with plan" |
| Community theme | Logged in (no/lower plan) | Grid & Details | Shows "Community" + "Available on Business" |
| Community theme | Logged in (Business plan) | Grid & Details | Shows "Community" + "Included with plan" |
| Included plan theme | Logged in (with plan) | Grid & Details | Shows "Included with plan" |


| Before | After |
| ---- | ---- |
| <video src="https://github.com/user-attachments/assets/b973e0ef-edda-4aec-8f0e-1d07336f0a37" controls width="250"></video> | <video src="https://github.com/user-attachments/assets/7dd32a91-3cbe-47e9-9742-bc4e17317e58" controls width="250"></video> |

## Pre-merge Checklist
- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] Have you written new tests for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

abotic committed:
Theme tier badges changes (LiTS & LoTS)

mikachan commented on client/components/theme-tier/theme-tier-badge/index.js:
```diff
@@ -23,22 +21,28 @@ export default function ThemeTierBadge( {
-	);
 	const themeTier = useThemeTierForTheme( themeId );
 	const isThemeAllowed = useIsThemeAllowedOnSite( siteId, themeId );
-	const isUpdatedBadgeDesign = useIsUpdatedBadgeDesign();
```
Removing any logic based on `isUpdatedBadgeDesign` makes sense to me based on [this comment](https://github.com/Automattic/wp-calypso/issues/95471#issuecomment-2593226443) 👍

abotic replied:
Thanks for linking this comment! It helps knowing that the right decision was made - as I was not 100% sure if this will be used in future.

mikachan commented on packages/design-picker/src/components/theme-card/index.tsx:
```diff
@@ -129,19 +123,9 @@ const ThemeCard = forwardRef(
-								/>
-							</Suspense>
+						{ ! isActive && badge && (
+							<div className="theme-card__info-badge-container">{ badge }</div>
```
I see that with these changes we've removed the link to the theme details page. I'm not sure how much value this is adding if the theme is already active, so I don't think it needs to be kept, but just highlighting that this is different compared to trunk:

![image](https://github.com/user-attachments/assets/6b083a2c-fbf0-4b1c-85a4-e235a3edd334)

abotic replied:
I went with an initial approach to keep consistency across all themes by removing everything except theme name and badge. Not sure was that a good approach in removing this as well.

@fditrapani Could you help us with this decision maybe? Thanks!

fditrapani replied:
I think there's still value in getting people to the details page. There are customization instructions there. When you're in the showcase, would it be possible to take people to the theme details page. From there, there's a button to customize. This mirrors the behaviour in Core.

mikachan reviewed (commented):
I'm having a go at reviewing this based on spending some time implementing [the new designs mentioned here](https://github.com/Automattic/wp-calypso/issues/95471#issuecomment-2529511388).

These changes look great so far when comparing them to the above designs, including all the different tiers.

One potential change I noticed is that I believe any free themes should always show a "free" badge, rather than the "included with plan" badge, e.g. TT5 and LeanCV in this screenshot:

![image](https://github.com/user-attachments/assets/e4ce3dd0-6fc5-4119-b486-ad602ffb944d)

mikachan commented:
I'd also like to highlight [this comment](https://github.com/Automattic/wp-calypso/pull/98435#issuecomment-2593985621) from @jeryj:

> It looks like the "Available on ____" badge work and removing tooltips has already been done behind the isGoalsAtFrontExperiment flag in https://github.com/Automattic/wp-calypso/pull/97655. If we're set on shipping this design, I think a lot of the work will just be to remove the isGoalsAtFrontExperiement.

I'm wondering if that should be part of this PR, or maybe it could be done in a separate branch?

abotic committed:
resolve conflicts

abotic commented:
> I'm having a go at reviewing this based on spending some time implementing [the new designs mentioned here](https://github.com/Automattic/wp-calypso/issues/95471#issuecomment-2529511388).
> 
> These changes look great so far when comparing them to the above designs, including all the different tiers.
> 
> One potential change I noticed is that I believe any free themes should always show a "free" badge, rather than the "included with plan" badge, e.g. TT5 and LeanCV in this screenshot:
> 
> ![image](https://private-user-images.githubusercontent.com/1645628/403826833-e4ce3dd0-6fc5-4119-b486-ad602ffb944d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1NzE4MzgsIm5iZiI6MTczNzU3MTUzOCwicGF0aCI6Ii8xNjQ1NjI4LzQwMzgyNjgzMy1lNGNlM2RkMC02ZmM1LTQxMTktYjQ4Ni1hZDYwMmZmYjk0NGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjJUMTg0NTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTc5YzU2ZmQwZjNhZGFkZTZmYjg2MmVhMzgyMjAxNWQ4NGUxNDEwNTk4NTc0ZWU2ZTZiODM1MDU0OTgxYjU0NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.JhmY44QaWGDzS3PLi92NiQXLnwaGzHARucnKyCTBIl8)

Thanks for taking a look @mikachan ! I was thinking about this exact point - the "Free" vs "Included" badge choice for Free themes.

The reasoning behind  "Included" badge that I went with was just to help users better appreciate the whole package when they are on a Personal+ plan, even if some themes would be free anyway.

But you raise a really good point about clarity. What's your take on which would be more intuitive for users? 
@fditrapani , do you maybe have thoughts on which approach we should go with?

fditrapani commented:
> @fditrapani , do you maybe have thoughts on which approach we should go with?

Thanks for bringing this up. It's a great point. I think it's worthwhile to delineate between the free themes and what's included in your plan. I support making that change!

abotic commented:
> I'd also like to highlight [this comment](https://github.com/Automattic/wp-calypso/pull/98435#issuecomment-2593985621) from @jeryj:
> 
> > It looks like the "Available on ____" badge work and removing tooltips has already been done behind the isGoalsAtFrontExperiment flag in #97655. If we're set on shipping this design, I think a lot of the work will just be to remove the isGoalsAtFrontExperiement.
> 
> I'm wondering if that should be part of this PR, or maybe it could be done in a separate branch?

@mikachan  Thanks for flagging this. I agree this should be handled in a separate PR to keep the current changes focused.

Also, found a bug with community theme badges that needs fixing in some underlying code and state. I'll have it fixed in the next commit and we can do another round of testing/review if possible.

fditrapani commented:
Thanks for putting this PR together, it's looking great! This might still be in progress but just wanted to point out that when I look at partner themes on a business site, it's saying they're included in my plan:

<img width="1456" alt="image" src="https://github.com/user-attachments/assets/6f1ae63c-f3d5-4a8d-8756-7d92aacd2938" />

In these cases, can we show the monthly price:
![image](https://github.com/user-attachments/assets/496d6ca4-e7ff-4c16-b440-8479eb2840bc)

fditrapani commented:
hey folks, just checking in to see how this is going and if there's anything I can do to unblock it if we're blocked?

abotic commented:
> hey folks, just checking in to see how this is going and if there's anything I can do to unblock it if we're blocked?

Hi @fditrapani , I was blocked on other task that took some time to resolve. Jumping on this one today and will try to get it ready as soon as possible.

fditrapani commented:
Awesome. Thanks for the update @abotic!

abotic committed:
Merge branch 'trunk' into update/theme-tier-badges-changes

abotic committed:
Fixes, changes and optimizations

- Fix partner theme price display logic for business plan
- Add proper theme bundle/partner price handling
- Refactor components with proper memoization and selector optimizations
- Clean up unused imports and tooltip logic
- Add proper equality checks to prevent unnecessary rerenders
- Fix theme upgrade/included badge conditional rendering

abotic committed:
Fixes, changes and optimizations

- Fix partner theme price display logic for business plan
- Add proper theme bundle/partner price handling
- Refactor components with proper memoization and selector optimizations
- Clean up unused imports and tooltip logic
- Add proper equality checks to prevent unnecessary rerenders
- Fix theme upgrade/included badge conditional rendering

abotic commented:
Hi @fditrapani  and @mikachan 👋 

Thank you for your comments, I've made some changes and it is ready for review.

Recent changes:
- Fixed all review comments
- Updated bundled/community/partner theme badge display
- Added price display for partner themes on Business plan
- Cleaned up unused code
- Refactored components for better performance with memoization and selector optimizations

Updated testing instructions/table and added new video showing free and business plan showcase (Personal and Premium follow same pattern).

Looking forward to your feedback.

fditrapani commented:
Thanks for making these updates @abotic! It's getting really close. I'll let you decide whether you want to merge this and work on the following tweaks as a follow up:

**Upgrade badge font**
The upgrade badge's font is incorrect. It's using Recoletta like the title but should be the standard sans-serif font we use for body copy:

<img width="1678" alt="image" src="https://github.com/user-attachments/assets/869853ad-10a9-4b5f-9cf5-1cf97ca867bc" />

here's how it should look:
<img width="1687" alt="image" src="https://github.com/user-attachments/assets/80551b5f-ad11-4c9b-9502-9d5ce45666bf" />

**Remove partner badge**
Now that we're presenting the plans and pricing better, the partner badge feels redundant. Can we remove it from the listings:

<img width="1669" alt="image" src="https://github.com/user-attachments/assets/8d436687-1891-4067-8978-d58d32c72a30" />

and the theme details page:
<img width="1684" alt="image" src="https://github.com/user-attachments/assets/da5acf5d-6fc6-43cf-8051-6b53251f16af" />

The same goes for the WooCommerce badge:
<img width="1684" alt="image" src="https://github.com/user-attachments/assets/2a3de4b4-3ffc-4fc7-8627-28ab6ccb7b75" />

We have a filter for that, so labelling it isn't as necessary.

jasmussen commented:
Nice work, all. If you'll forgive my passer-by comment, I wonder about the font-weight here:

![Screenshot 2025-02-04 at 08 45 59](https://github.com/user-attachments/assets/50e0335b-27d3-4801-80e6-4b4703a25a5d)

Potentially this is just a "draw on top of" mockup that's imprecise, or because it's white on black. But it's looking a little thin to me. Perhaps it _is_ the same, but if it isn't, I'd expect it to inherit the same font weight as this recommended tab:

![Screenshot 2025-02-04 at 08 47 04](https://github.com/user-attachments/assets/6a592ce2-218f-41e3-892e-4f8c9865b25c)

abotic committed:
remove woocommerce and partner badge and apply font styles to badge labels

abotic commented:
@fditrapani @jasmussen Thanks for the great comments.

I've committed some changes - please let me know if additional changes are needed:

- Upgraded the badge font and applied better styles
- Removed `Partner` and `WooCommerce` badges. Please note that only two badges now remain: `Community` and `Sensei LMS` (Course theme)

Regarding the font weight that @jasmussen mentioned: both the badge label and recommended tab have a fixed font weight of 400. Maybe the overall styling made it look thin. If it still doesn't look great, we could set it to 500.

abotic committed:
update snapshot tests

abotic committed:
Merge remote-tracking branch 'origin/trunk' into update/theme-tier-badges-changes

abotic committed:
fix build

abotic committed:
fix build-2

abotic committed:
use hardcoded font-family

jasmussen commented on packages/components/src/theme-type-badge/bundled-badge/style.scss:
```diff
@@ -15,7 +15,7 @@
 		white-space: nowrap;
 		font-size: 0.875rem;
-		font-family: $sans;
+		font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
```
Could this be `inherit`? Or is there a variable font stack we can use rather than the hardcoded value? Just curious!

abotic replied:
Unfortunately, we can't use `inherit` (it would work great in theme showcase grid, but not on details.)
I've used hardcoded values as it was used like that on other places too, but actually there is really no reason why we would not import `@automattic/typography/styles/variables` and make it cleaner.

Fixed it, thank you!

jasmussen replied:
Nice, thank you! Good improvement. Although that system font stack is unlikely to drift over time, it's nice to have it "indexed", as it were.

abotic committed:
use typography instead of hardcoded css values

fditrapani reviewed (approved):
This looks great now! Thanks for the updates ⭐

xavier-lc commented on packages/design-picker/src/components/theme-card/index.tsx:
```diff
@@ -129,20 +124,13 @@ const ThemeCard = forwardRef(
-						{ ! isActive && badge && <>{ badge }</> }
-						{ optionsMenu && <div className="theme-card__info-options">{ optionsMenu }</div> }
 						{ isActive && <ActiveBadge /> }
+						{ optionsMenu && optionsMenu && (
```
```suggestion
						{ optionsMenu && (
```

abotic replied:
Nice catch :)

xavier-lc commented on client/components/theme-tier/theme-tier-badge/index.js:
```diff
@@ -23,49 +24,77 @@ export default function ThemeTierBadge( {
-	const getBadge = () => {
-		if ( isUpdatedBadgeDesign && 'free' === themeTier?.slug ) {
+
+	const getBadge = useMemo( () => {
```
```suggestion
	const badge = useMemo( () => {
```

I'd rename the function, since having it as `getBadge` suggests that it's a getter that needs to be called even after the `useMemo` in order to get the value of the badge. Wdyt?

abotic replied:
Agree! Thanks

xavier-lc reviewed (commented):
Looks good, two minor comments.

abotic committed:
minor refactor

xavier-lc reviewed (approved):
LGTM!

mmtr commented:
Is it expected that badges are not consistent between the logged-in Theme Showcase and the onboarding's Design Picker?

Page | Before | After
--- | --- | ---
Theme Showcase | <img width="478" alt="Screenshot 2025-02-10 at 15 57 14" src="https://github.com/user-attachments/assets/2cc33174-bf89-4937-be3d-f7b41be7c4c6" /> | <img width="482" alt="Screenshot 2025-02-10 at 15 55 29" src="https://github.com/user-attachments/assets/dd274305-7417-4be8-ac42-58e5e3e2664c" />
Design Picker | <img width="406" alt="Screenshot 2025-02-10 at 15 45 40" src="https://github.com/user-attachments/assets/843d7bfe-a9c2-4c9d-9668-6945313aa6da" /> | <img width="403" alt="Screenshot 2025-02-10 at 15 45 43" src="https://github.com/user-attachments/assets/feacd385-a091-4371-b784-230b556d4539" />

fditrapani commented:
Good catch @mmtr. That looks like a regression. They should be the same.

dsas commented:
@abotic, popping this back out of Ready To merge for the [inconsistency](https://github.com/Automattic/wp-calypso/pull/98451#issuecomment-2648280314) to be looked at

abotic committed:
apply same styles on onboarding badges

abotic commented:
@fditrapani @mmtr Same styling applied on onboarding as well. Thanks for feedback!

Please note that there was one minor regression issue with plans that had pricing shown near them.
There seems to be too much text in badge when we are previewing a theme while we are on onboarding.

We could change the font, apply some margins while being onboarded or something else, but changing the "plus" to "+" seemed to me like a best option to keep exactly same styling between onboarding and logged-in theme picker. If this is not okay, please let me know some other idea we could apply here. 

Example:
| Before | After |
| ---- | ---- |
| ![Screenshot 2025-02-14 at 15 46 52](https://github.com/user-attachments/assets/a5daabab-3b17-4c4e-9c69-b5ed6b34b67b) | ![Screenshot 2025-02-14 at 15 46 25](https://github.com/user-attachments/assets/2a0d2d7b-ed73-409e-b3ed-deea2cb1d2c9) |

fditrapani commented:
Thanks for the updates. We can save even more space, for this specific case, if we update it to "On Business + $10.00/month"

mmtr commented:
Maybe a design nitpick, but I noted that the badges in the Design Picker look bigger now because the font-size has been increased to 14px:

<img width="1253" alt="Screenshot 2025-02-17 at 16 05 54" src="https://github.com/user-attachments/assets/a3d1a2e4-2489-48f0-8122-9f7753bd99f6" />

<img width="1271" alt="Screenshot 2025-02-17 at 16 04 28" src="https://github.com/user-attachments/assets/64a96190-dab0-4213-80a9-056297028624" />

Currently, in production, the font-size is set to 12px which also matches the font-size used in the Figma design I can find in p9Jlb4-fea-p2.

This makes think that the font-size increase has not been intentional.

mmtr commented:
Community themes also show a weird layout, is this expected?
<img width="1080" alt="Screenshot 2025-02-17 at 16 16 08" src="https://github.com/user-attachments/assets/212a2fdc-9877-489d-a5f6-b3165fe3410f" />

mmtr commented:
Another issue is that the theme name may be hidden by badges:

Before | After
--- | ---
<img width="352" alt="Screenshot 2025-02-17 at 16 20 06" src="https://github.com/user-attachments/assets/806438e2-4e4b-4a0f-b3c8-31212f218d9b" /> | <img width="362" alt="Screenshot 2025-02-17 at 16 20 11" src="https://github.com/user-attachments/assets/eb154474-eb34-4929-b9b7-ce28518731b7" />

abotic committed:
Fix font size, remove additional badges from theme showcase, leaving them in details only

abotic committed:
Merge remote-tracking branch 'origin' into update/theme-tier-badges-changes

abotic committed:
fix styling of free and included tiers

abotic commented:
@mmtr @fditrapani  Thanks for great regression testing!
Everything discussed since the last commit has been fixed/changed.

That includes issue with a fonts, overlapping of elements and extra badges. We have now removed extra badges (community, bundle, WooCommerce) from theme showcase, but they are still available in theme details.  

Please let me know if you find anything else.

fditrapani commented:
Thanks for these updates @abotic. We're almost there! I did one final pass on everything across the `/themes`, onboarding, and the theme marketplace within a site. The only thing I found now is that when you're on a business plan, we show the price in a black pill:

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/b5f31c8d-8f67-4616-b052-b15057edf899" />

Can we show it as regular text with the price like all the others:
<img width="1552" alt="image" src="https://github.com/user-attachments/assets/211876bf-0922-4781-a5d2-84c91574a91a" />

The same on the theme details, we only show the price in the dark pill:
<img width="1552" alt="image" src="https://github.com/user-attachments/assets/b56e98e7-f3c7-4b73-8436-a4535a5e9fbc" />

Can we add a star in there like all the other upgrades and add some text to not make it feel so transactional since we have the space :) 
<img width="1552" alt="image" src="https://github.com/user-attachments/assets/86774761-669e-4176-b0ed-96dd0ca2d61b" />

abotic committed:
Update partner badge and fix font

abotic committed:
update snapshot test for theme showcase

abotic committed:
add small gap when there are multiple badges

abotic committed:
Merge remote-tracking branch 'origin/trunk' into update/theme-tier-badges-changes

abotic commented:
@fditrapani Everything you proposed regarding pricing has been implemented and pushed.
There were a few minor bugs with font sizes in the included and free plans getting overridden in some edge cases, so I fixed those as well.

Let me know if there's anything else you would like to change/improve, and I can jump on it immediately :)

abotic committed:
Merge remote-tracking branch 'origin/trunk' into update/theme-tier-badges-changes

fditrapani reviewed (approved):
Thank you for all the updates @abotic! This looks great now.

mmtr commented on client/components/theme-tier/theme-tier-badge/index.js:
```diff
@@ -23,47 +24,81 @@ export default function ThemeTierBadge( {
-		isThemePurchased( state, themeId, siteId )
+	const themeType = useSelector(
+		( state ) => getThemeType( state, themeId ),
+		( prev, next ) => prev === next
```
Is this needed? I always thought this was the default behavior:

> `useSelector()` only forces a re-render if the selector result appears to be different than the last result. The default comparison is a strict `===` reference comparison.
> – https://react-redux.js.org/api/hooks#equality-comparisons-and-updates

abotic replied:
Yes, you're absolutely right! The custom equality function `(prev, next) => prev === next` is redundant since it's identical to useSelector's default behavior. 

I'll remove it and simplify to just:
```
const themeType = useSelector(state => getThemeType(state, themeId));
```
Thanks for catching this!

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-community-badge.js:
```diff
@@ -1,86 +1,40 @@
-	const isEcommerceTrialMonthly = planSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
-	const planTooltipName = isEcommerceTrialMonthly ? 'ecommerce' : 'business';
 
-	const getTooltipMessage = () => {
-		if ( ! isEcommerceTrialMonthly ) {
-			return createInterpolateElement(
-				translate(
-					'This community theme can only be installed if you have the <Link>%(businessPlanName)s plan</Link> or higher on your site.',
-					{ args: { businessPlanName: getPlan( PLAN_BUSINESS )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
-			);
-		}
-		if ( isEcommerceTrialMonthly ) {
-			return createInterpolateElement(
-				translate(
-					"This theme can't be installed on a trial site. Please upgrade to the <Link>%(ecommercePlanName)s plan</Link> to install this theme.",
-					{ args: { ecommercePlanName: getPlan( PLAN_ECOMMERCE )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
-			);
-		}
-	};
```
This causes a confusing badge now for trial plans.

For instance, on a site with an Commerce trial plan, we now say the theme is available on the Business plan, but they cannot really buy that plan (they can only buy the Commerce non-trial plan):

Before | After
--- | ---
<img width="453" alt="Screenshot 2025-02-19 at 19 54 33" src="https://github.com/user-attachments/assets/5ace1549-65bc-4003-b228-ad6e8a60a002" /> | <img width="384" alt="Screenshot 2025-02-19 at 19 54 41" src="https://github.com/user-attachments/assets/510e448a-9df7-4395-a099-243fca3fb774" />

On these cases, we should say "Available on the Commerce plan" instead of "Available on the Business plan".

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:
```diff
@@ -1,172 +1,84 @@
-					{ args: { businessPlanName: getPlan( PLAN_BUSINESS )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
+	const priceBadge = useMemo( () => {
+		if ( ! isThemeAllowed ) {
+			return (
+				<ThemeTierPlanUpgradeBadge
+					showPartnerPrice={ showPartnerPrice }
+					hideBackgroundOnUpgrade={ hideBackgroundOnUpgrade }
+				/>
 			);
 		}
 
-		if ( isPartnerThemePurchased && ! isThemeAllowed && isEcommerceTrialMonthly ) {
-			return createInterpolateElement(
-				translate(
-					"You have a subscription for this theme, but it isn't usable on a trial plan site. Please upgrade to the <link>%(ecommercePlanName)s plan</link> to install this theme.",
-					{ args: { ecommercePlanName: getPlan( PLAN_ECOMMERCE )?.getTitle() ?? '' } }
-				),
-				{
-					Link: <ThemeTierBadgeCheckoutLink plan={ planTooltipName } />,
-				}
-			);
-		}
```
Same for partner themes and Commerce trial sites. We cannot ask them to buy the Business plan, because their only option is the Commerce plan:

Before | After
--- | ---
<img width="444" alt="Screenshot 2025-02-19 at 19 59 11" src="https://github.com/user-attachments/assets/f7bfb919-3429-416f-bc85-48320e20fa38" />  | <img width="387" alt="Screenshot 2025-02-19 at 19 59 18" src="https://github.com/user-attachments/assets/c21cea9d-faab-4fa5-afc7-fe57ad0d25b6" />

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-upgrade-badge.js:
```diff
@@ -3,69 +3,65 @@ import { PremiumBadge } from '@automattic/components';
-			// Use shorter version if full label is too long
-			const MAX_LABEL_LENGTH = 45;
 			return fullLabel.length > MAX_LABEL_LENGTH
-				? /* translators: This is a shorter version of the text "Available on %(planName)s plus %(price)s/month". */
```
We should keep this comment since it's helpful for translators.

mmtr reviewed (commented):
One thing that caught my attention is that it seems that the theme title and the badge are not aligned (I added some read/blue lines to better note it):

<img width="612" alt="Screenshot 2025-02-19 at 19 30 02" src="https://github.com/user-attachments/assets/4fe8b224-2eb3-400e-9bbe-5fe88d48076c" />

Feel free to ignore this feedback if you believe this is something not important.

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:
```diff
@@ -1,172 +1,84 @@
+		translate,
+	] );
 
-			<PremiumBadge
-				className="theme-tier-badge__content is-third-party"
-				focusOnShow={ false }
-				isClickable={ false }
-				labelText={ translate( 'Partner' ) }
-				shouldHideIcon
-				tooltipClassName="theme-tier-badge-tooltip"
-				tooltipContent={ partnerTooltipContent }
-				tooltipPosition="top"
-			/>
```
Is it expected that the "Partner" badge is now not displayed in the theme details view?

Before | After
--- | ---
<img width="474" alt="Screenshot 2025-02-20 at 11 22 43" src="https://github.com/user-attachments/assets/5ab23919-2ca1-4c59-b678-b2f1d32c7106" /> | <img width="408" alt="Screenshot 2025-02-20 at 11 23 02" src="https://github.com/user-attachments/assets/782dc97e-32fc-4184-83cd-5cad3b26fbac" />


Asking because this is inconsistent with the "Community" or "WooCommerce" badges:

Before | After
--- | ---
<img width="216" alt="Screenshot 2025-02-20 at 11 36 25" src="https://github.com/user-attachments/assets/160ffe44-c866-4aa4-b9af-3dbfcfb72b84" /> | <img width="284" alt="Screenshot 2025-02-20 at 11 36 40" src="https://github.com/user-attachments/assets/b63b0062-1115-413e-9c9d-83e17bb85568" />
<img width="271" alt="Screenshot 2025-02-20 at 11 37 24" src="https://github.com/user-attachments/assets/cc7f1f7e-122a-43ac-a8de-6171cc8d5c35" /> | <img width="296" alt="Screenshot 2025-02-20 at 11 37 29" src="https://github.com/user-attachments/assets/47136276-9346-494d-95d1-2627f3a203e4" />

mmtr commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -472,6 +472,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 	const getBadge = ( themeId: string, isLockedStyleVariation: boolean ) => (
 		<ThemeTierBadge
 			canGoToCheckout={ false }
+			isThemeShowcase
```
Nitpick (non-blocking): This is a bit confusing, because the `unified-design-picker.tsx` file is part of the Design Picker, not the Theme Showcase.

What are you trying to achieve when `isThemeShowcase` is true? It seems that the intention is to support two different styles, one for when there is a list of themes and another one for when only one theme is visible.

Maybe you could rename the prop to something else like `isSingleTheme` or `isThemeList`, since that's generic enough for both the Design Picker (`/setup/onboarding/designSetup`) and the Theme Showcase (`/themes`).

abotic committed:
Fix trial commerce plan and resolve other comments

abotic committed:
fix minor alignment issue

abotic committed:
update snapshot test

abotic committed:
Merge remote-tracking branch 'origin/trunk' into update/theme-tier-badges-changes

abotic committed:
Fix merging issues and missing params

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-bundled-badge.js:
```diff
@@ -1,82 +1,42 @@
 	const bundleSettings = useBundleSettingsByTheme( themeId );
 	const isThemeIncluded = useSelector(
-		( state ) => siteId && canUseTheme( state, siteId, themeId )
+		( state ) => siteId && canUseTheme( state, siteId, themeId ),
+		( prev, next ) => prev === next
```
Nitpick (non-blocking): This is unnecessary, since it's the default behavior
```suggestion
		( state ) => siteId && canUseTheme( state, siteId, themeId )
```

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:
```diff
@@ -1,172 +1,99 @@
-	const isPartnerThemePurchased = useSelector( ( state ) =>
-		siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false
+	const isPartnerThemePurchased = useSelector(
+		( state ) => ( siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false ),
+		( prev, next ) => prev === next
```
Nitpick (non-blocking): This is unnecessary, since it's the default behavior
```suggestion
		( state ) => ( siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false )
```

mmtr commented on client/components/theme-tier/theme-tier-badge/theme-tier-community-badge.js:
```diff
@@ -1,86 +1,40 @@
-			<div data-testid="upsell-message">{ getTooltipMessage() }</div>
-		</>
+	const isThemeIncluded = useSelector(
+		( state ) => siteId && canUseTheme( state, siteId, themeId ),
+		( prev, next ) => prev === next
```
Nitpick (non-blocking): This is unnecessary, since it's the default behavior
```suggestion
		( state ) => siteId && canUseTheme( state, siteId, themeId )
```

mmtr reviewed (approved):
Lovely work @abotic! Thanks for addressing all the feedback, I couldn't find any regression now. 🚀

abotic committed:
simplify redudant selectors

abotic committed:
Merge remote-tracking branch 'origin/trunk' into update/theme-tier-badges-changes

abotic commented:
Thanks @mmtr !
I’ve fixed those comments regarding selectors and merged trunk into the branch. Everything seems stable on my end.
If you’d like to take another look, please let me know.

mmtr commented:
> If you’d like to take another look, please let me know.

Not necessary, the last changes don't seem to be dangerous, so if you feel confident you can merge without asking for another review.
